pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -142,31 +142,65 @@ func testHTTPStats(t *testing.T, aggregateByStatusCode bool) {
 }
 
 func TestHTTPSViaLibraryIntegration(t *testing.T) {
-	if !httpSupported(t) {
-		t.Skip("HTTPS feature not available on pre 4.14.0 kernels")
-	}
 	if !httpsSupported(t) {
 		t.Skip("HTTPS feature not available/supported for this setup")
 	}
 
+	buildPrefetchFileBin(t)
+
 	tlsLibs := []*regexp.Regexp{
 		regexp.MustCompile(`/[^\ ]+libssl.so[^\ ]*`),
 		regexp.MustCompile(`/[^\ ]+libgnutls.so[^\ ]*`),
 	}
 	tests := []struct {
-		name     string
-		fetchCmd []string
+		name         string
+		fetchCmd     []string
+		prefetchLibs []string
 	}{
-		{name: "wget", fetchCmd: []string{"wget", "--no-check-certificate", "-O/dev/null"}},
-		{name: "curl", fetchCmd: []string{"curl", "--http1.1", "-k", "-o/dev/null"}},
+		{
+			name:     "wget",
+			fetchCmd: []string{"wget", "--no-check-certificate", "-O/dev/null"},
+		},
+		{
+			name:     "curl",
+			fetchCmd: []string{"curl", "--http1.1", "-k", "-o/dev/null"},
+		},
+	}
+
+	for _, test := range tests {
+		fetch, err := exec.LookPath(test.fetchCmd[0])
+		if err != nil {
+			t.Skipf("%s not found; skipping test.", test.fetchCmd)
+		}
+		ldd, err := exec.LookPath("ldd")
+		if err != nil {
+			t.Skip("ldd not found; skipping test.")
+		}
```

Comment:
I think this could be moved up one more layer and only done once.

Reply:
great catch, thanks